core/trino-main/src/main/java/io/trino/server/testing/shuffle/LocalFileSystemShuffleService.java

```diff
@@ -0,0 +1,499 @@
+            Path taskOutputBasePath = getTaskOutputBasePath(taskPartitionId);
+            try {
+                List<Path> attemptPaths = Files.list(taskOutputBasePath)
+                        .filter(Files::isDirectory)
+                        .collect(toImmutableList());
+                checkState(!attemptPaths.isEmpty(), "no attempts found for %s", taskPartitionId);
+
+                List<Path> committedAttemptPaths = attemptPaths.stream()
+                        .filter(LocalFileSystemShuffle::isCommitted)
+                        .collect(toImmutableList());
+                checkState(!committedAttemptPaths.isEmpty(), "no committed attempts found for %s", taskPartitionId);
+
+                return committedAttemptPaths.get(0);
+            }
+            catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
+        }
+
+        private Path getTaskOutputBasePath(int taskPartitionId)
+        {
+            return getShufflePath().resolve(Integer.toString(taskPartitionId));
+        }
+
+        private Path getShufflePath()
+        {
+            return baseDirectory.resolve(shuffleHandle.getShuffleId());
+        }
+
+        private static boolean isCommitted(Path attemptPath)
+        {
+            Path commitMarkerFilePath = attemptPath.resolve("committed");
+            return Files.exists(commitMarkerFilePath);
+        }
+
+        private static Map<Integer, Path> getCommittedPartitions(Path committedAttemptPath)
+        {
+            try {
+                List<Path> partitionFiles = Files.list(committedAttemptPath)
+                        .filter(path -> path.toString().endsWith(".data"))
+                        .collect(toImmutableList());
+                ImmutableMap.Builder<Integer, Path> result = ImmutableMap.builder();
+                for (Path partitionFile : partitionFiles) {
+                    Matcher matcher = PARTITION_FILE_NAME_PATTERN.matcher(partitionFile.getFileName().toString());
+                    checkState(matcher.matches(), "unexpected partition file: %s", partitionFile);
+                    int partitionId = Integer.parseInt(matcher.group(1));
+                    result.put(partitionId, partitionFile);
+                }
+                return result.build();
+            }
+            catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
+        }
+
+        @Override
+        public void registerLostTaskOutputListener(ShuffleLostTaskOutputListener listener)
+        {
+        }
+
+        @Override
+        public CompletableFuture<List<ShufflePartitionHandle>> getInputPartitionHandles()
+        {
+            return inputPartitionHandles;
+        }
+
+        @Override
+        public CompletableFuture<?> isInputReady()
+        {
+            return inputReady;
+        }
+
+        @Override
+        public void close()
+        {
+            try {
+                deleteRecursively(getShufflePath());
```

Comment:
When I run unit tests like `TestLocalFileSystemShuffleService` locally, I have to pass the option `RecursiveDeleteOption.ALLOW_INSECURE`. Otherwise I will get `InsecureRecursiveDeleteException`